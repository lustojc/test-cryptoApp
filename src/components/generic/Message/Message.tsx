interface props {
  color: string;
  message: string;
  fontSize: number;
  backgroundColor?: string;
  fontStyle?: 'italic' | 'bold' | 'normal' | 'oblique';
}

export default function Message({ color, message, fontSize, backgroundColor, fontStyle }: props) {
  const style = {
    color,
    fontSize: fontSize,
    display: 'inline',
    padding: '10px',
    backgroundColor,
    fontStyle,
  };

  return <div style={style}>{message}</div>;
}
