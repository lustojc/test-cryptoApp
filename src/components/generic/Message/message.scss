.message {
  &-fontSize {
    &__normal {
      font-size: 14;
    }
  }

  &-color {
    &__black {
      color: black;
    }
    &__red {
      color: red;
    }

    &__green {
      color: green;
    }
  }
}
