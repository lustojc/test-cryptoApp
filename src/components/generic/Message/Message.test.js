import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';

import Message from './Message';

describe('test message component', () => {
  it('render message', () => {
    render(<Message message="Correct value" backgroundColor="red" />);

    const messageElement = screen.getByText(/correct value/i);
    expect(messageElement).toBeInTheDocument();
    expect(messageElement).toHaveStyle({ backgroundColor: 'red' });
    expect(messageElement).toMatchSnapshot();
  });
});
