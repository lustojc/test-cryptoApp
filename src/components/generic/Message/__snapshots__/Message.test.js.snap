// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test message component render message 1`] = `
<div
  data-cy="errorMessage"
  style="display: inline; padding: 10px; background-color: red;"
>
  Correct value
</div>
`;
